lib/ASTGen/Sources/ASTGen/Decls.swift

```diff
@@ -1,162 +1,503 @@
 import CASTBridging
-import SwiftParser
 import SwiftSyntax
 
+// MARK: - TypeDecl
+
 extension ASTGenVisitor {
   public func visit(_ node: TypealiasDeclSyntax) -> ASTNode {
-    let aliasLoc = bridgedSourceLoc(for: node.typealiasKeyword)
-    let equalLoc = bridgedSourceLoc(for: node.initializer.equal)
-    var nameText = node.identifier.text
-    let name = nameText.withBridgedString { bridgedName in
-      return ASTContext_getIdentifier(ctx, bridgedName)
-    }
-    let nameLoc = bridgedSourceLoc(for: node.identifier)
-    let genericParams = node.genericParameterClause.map { self.visit($0).rawValue }
-    let out = TypeAliasDecl_create(
-      self.ctx, self.declContext, aliasLoc, equalLoc, name, nameLoc, genericParams)
+    let (name, nameLoc) = self.bridgedIdentifierAndSourceLoc(for: node.identifier)
 
-    let oldDeclContext = declContext
-    declContext = out.declContext
-    defer { declContext = oldDeclContext }
+    return .decl(
+      TypeAliasDecl_create(
+        self.ctx,
+        self.declContext,
+        self.bridgedSourceLoc(for: node.typealiasKeyword),
+        name,
+        nameLoc,
+        node.genericParameterClause.map(self.visit)?.rawValue,
+        self.bridgedSourceLoc(for: node.initializer.equal),
+        self.visit(node.initializer.value).rawValue,
+        node.genericWhereClause.map(self.visit)?.rawValue
```

Comment:
Sounds good. Would you mind if I make this overload generic (and always-inline) to avoid duplication?

Reply:
If you have an idea of how to make that work generically, I definitely don’t mind. I thought it wouldn’t work because what would you constrain the generic type by? But I’m open for ideas.

Reply:
Never mind, I thought of constraining it to `SyntaxProtocol`, but that implies a wasteful call to the dispatch method. Besides, the result need not be an `ASTNode`.